packages/core/src/telemetry/clearcut-logger/clearcut-logger.ts

```
@@ -142,8 +167,33 @@ export class ClearcutLogger {
         if (this.config?.getDebugMode()) {
           console.log('Clearcut POST request error: ', e);
         }
-        // Add the events back to the front of the queue to be retried.
-        this.events.unshift(...eventsToSend);
+        // Add the events back to the front of the queue to be retried, but limit retry queue size
+        const eventsToRetry = eventsToSend.slice(-this.max_retry_events); // Keep only the most recent events
+
+        // Add retry events to the front of the deque (O(1) operations)
+        // If the queue is full, unshift will throw, and we'll stop requeueing.
+        for (let i = eventsToRetry.length - 1; i >= 0; i--) {
+          try {
+            this.events.unshift(eventsToRetry[i]);
+          } catch (e) {
+            // This can happen if the queue is full, possibly due to a race condition with enqueueLogEvent.
```

please refactor avoid this catch block. there should be ways to avoid any race condition here. if there are failure modes I would much rather have us check that this.events is empty that catch to swallow exceptions.

---

I've refactored the code to avoid the try-catch block and prevent race conditions. The new  implementation:

1. Pre-calculates available space before attempting to requeue events
2. Uses deterministic size checks instead of relying on exception handling
3. Prioritizes the most recent events when the queue can't fit all retry events
4. Double-checks available space before each individual insert to handle concurrent enqueueLogEvent  calls

This approach eliminates the race condition by checking constraints before operations rather than  catching overflow exceptions.